src/dotnet/ToolPackage/PackageId.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.DotNet.InternalAbstractions;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class PackageId : IEquatable<PackageId>, IComparable<PackageId>
```

Comment:
This should be a struct to achieve the type safety without unnecessary allocation and pointer indirection

Reply:
Good point, will fix.

Reply:
Pushed.